### PR TITLE
[ceph] Another filter for 'ceph report'

### PR DESCRIPTION
**Pull request by pponnuvel**
Older sosreports may have the extra 'report' line as
the first line of 'ceph report' which makes it an invalid
json file. Added a filter that removes the first line if
it starts with 'report'.

Fixes: #353

Signed-off-by: Ponnuvel Palaniyappan <pponnuvel@gmail.com>

### CODE REVIEW

**Comment by pponnuvel** [comment]
Noticed the same 'first line' <-> 'last line' swap in few other files too. So extended it a bit.

File: hotsos/core/host_helpers/cli.py [hunk 1]
```diff
@@ -395,7 +396,12 @@ def format_json_contents(self, *args, **kwargs):  # pylint: disable=W0613
         with open(self.path) as f:
             lines = f.readlines()
 
-        if lines and lines[-1].startswith(self.last_line_filter):
+        if self.first_line_filter:
+            line_filter = self.first_line_filter
```
Comment (hemanthnakkina) [inline hotsos/core/host_helpers/cli.py:400]: Will it be simple to use python inbuilt function string strip() to handle this?
strip() can remove both leading and trailing characters.

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]
Please see inline comment

File: hotsos/core/host_helpers/cli.py (hunk 1 above)
Comment (pponnuvel) [reply hotsos/core/host_helpers/cli.py:400]: I am not sure I understand.

Do you mean loading the entire file into a string and then removing the parts from either end and writing that string back?

What we try to do here is remove the first or last line which looks like in either form:
```
report xyz
{
 # json formatted output
}
```
or 
```
{
 # json formatted output
}
report xyz
```

The extra line "report xyz" makes invalid json and parsing fails. So removing that extra line makes it work.

**Comment by hemanthnakkina** [comment]
@pponnuvel ignore my comment. I thought strip() will remove substrings but the function takes characters and strip the characters.  I am thinking if you can do substring and indexing to remove first or last line which will avoid creating temporary file.

**Pull request merged**